parser/fsimporter.go

```diff
@@ -732,6 +756,188 @@ func (fs *FSImporter) parseFiles(indexedFiles []*fpt.IndexedFile, parsingThreads
+								bytes := int64(origIPBytes + respIPBytes)
+								protocol := parseConn.Proto
+								service := parseConn.Service
+								dstPort := parseConn.DestinationPort
+								uid := parseConn.UID
+
+								var tuple string
+								if service == "" {
+									tuple = strconv.Itoa(dstPort) + ":" + protocol + ":-"
+								} else {
+									tuple = strconv.Itoa(dstPort) + ":" + protocol + ":" + service
+								}
+
+								// Safely store the number of conns for this uconn
+								mutex.Lock()
+
+								// Check if the map value is set
+								if _, ok := hostMap[srcKey]; !ok {
+									// create new host record with src and dst
+									hostMap[srcKey] = &host.Input{
+										Host:    srcUniqIP,
+										IsLocal: util.ContainsIP(fs.GetInternalSubnets(), srcIP),
+										IP4:     util.IsIPv4(src),
+										IP4Bin:  util.IPv4ToBinary(srcIP),
+									}
+								}
+
+								// Check if the map value is set
+								if _, ok := hostMap[dstKey]; !ok {
+									// create new host record with src and dst
+									hostMap[dstKey] = &host.Input{
+										Host:    dstUniqIP,
+										IsLocal: util.ContainsIP(fs.GetInternalSubnets(), dstIP),
+										IP4:     util.IsIPv4(dst),
+										IP4Bin:  util.IPv4ToBinary(dstIP),
+									}
+								}
+
+								// Check if the map value is set
+								if _, ok := uconnMap[srcDstKey]; !ok {
+									// create new uconn record with src and dst
+									// Set IsLocalSrc and IsLocalDst fields based on InternalSubnets setting
+									// we only need to do this once if the uconn record does not exist
+									uconnMap[srcDstKey] = &uconn.Input{
+										Hosts:      srcDstPair,
+										IsLocalSrc: util.ContainsIP(fs.GetInternalSubnets(), srcIP),
+										IsLocalDst: util.ContainsIP(fs.GetInternalSubnets(), dstIP),
+									}
+
+									// Can do this even if the connection is open. For each set of logs we
+									// process, we increment these values once per unique connection.
+									// This might mean that an open connection has caused these values
+									// to be incremented, but that is ok.
+									hostMap[srcKey].CountSrc++
+									hostMap[dstKey].CountDst++
+								}
+
+								// If the ConnStateList map doesn't exist for this entry, create it.
+								if uconnMap[srcDstKey].ConnStateList == nil {
+									uconnMap[srcDstKey].ConnStateList = make(map[string]*uconn.ConnState)
+								}
+
+								// If an entry for this open connection is present, first check
+								// to make sure it hasn't been marked as closed. If it was marked
+								// as closed, that supersedes any open connections information.
+								// Otherwise, if it's open then check if this more up-to-date data
+								// (i.e., longer duration)
+								if _, ok := uconnMap[srcDstKey].ConnStateList[uid]; ok {
+									if uconnMap[srcDstKey].ConnStateList[uid].Open {
+
+										if duration > uconnMap[srcDstKey].ConnStateList[uid].Duration {
```

Comment:
Mind collapsing these two conditions?

Reply:
Good call. In the initial version there was more happening in there but with this alternative version it's a bit unnecessary to have them be separate statements.